Actions/ReadSecrets/ReadSecretsHelper.psm1

```diff
@@ -52,42 +54,61 @@ function GetGithubSecret {
     if ($script:gitHubSecrets.PSObject.Properties.Name -eq $secret) {
         $value = $script:githubSecrets."$secret"
         if ($value) {
-            MaskValue -key $secret -value $value
-            return $value
+            MaskValue -key $envVar -value $value
+            if ($encrypted) {
```

Comment:
Consider adding tests for this part (encrypted secrets).

I see `ReadSecrets` action isn't very well tested. It's a good occasion to add more tests.

Reply:
Agree that CI tests of ReadSettings and ReadSecrets are not sufficient - they are primarily tested through the E2E tests.
I will work on adding test coverage of these two actions in a separate PR.